### PR TITLE
[dev] Add include versions for Dune builds.

### LINKED ISSUE #9537: dune and coqtop.byte
#### Description of the problem

I was trying to use coqtop.byte built with dune to debug something in equations and failed.
First issue is that just running coqtop.byte failed, one has to 
```
export LD_LIBRARY_PATH=PATH_TO_COQ/_build/install/default//lib/stublibs:$LD_LIBRARY_PATH
```
For this to run, otherwise the dynamic library linker fails to find a DLL.

After this, processing works, and I can drop to the toplevel, however `#use "include";;` fails, and after adding the PATH_TO_COQ/dev directory to the source files it succeeds but cannot find the object files it needs. Adding #directory "PATH_TO_COQ/_build/default/dev" it still fails with:
```
File "/Users/sozeau/dev/coq/coq/_build/default/dev/top_printers.ml", line 13, characters 5-10:
Error: Unbound module Sorts
Hint: Did you mean Sort?
File "/Users/sozeau/dev/coq/coq/_build/default/dev/vm_printers.ml", line 2, characters 5-9:
Error: Unbound module Term
Unbound value ppid.
...
```

### PR DESCRIPTION
**Pull request by ejgallego**
Fixes #9537

This way, users can do:
```
dune exec coqtop.byte
> Drop.
 # #directory "dev";;
 # #use "include_dune";;
```

### CODE REVIEW

File: dev/include_dune [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+   Alternatively, you can avoid typing #use "include" after each Drop
+   by adding the following lines in your $HOME/.ocamlinit :
+
+   #directory "+compiler-libs";;
+   if Filename.basename Sys.argv.(0) = "coqtop.byte"
+   then ignore (Toploop.use_silently Format.std_formatter "dev/include")
+*)
+
+(* For OCaml 3.10.x:
```
Comment (Zimmi48) [inline dev/include_dune]: Seems like this was copied from somewhere else. You can probably delete this comment.

Comment (ejgallego) [reply dev/include_dune]: I will indeed refactor a bit more.

**Comment by ejgallego** [comment]
That's ready, sorry I forgot to remove the tag.

File: dev/include_dune [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+(* Typical usage :
+
+   $ dune exec coqtop.byte # or even better : rlwrap coqtop.byte
+   Coq < Drop.
+   # #directory "dev";;
+   # #use "include";;
+
+   Alternatively, you can avoid typing #use "include" after each Drop
+   by adding the following lines in your $HOME/.ocamlinit :
```
Comment (Zimmi48) [inline dev/include_dune]: Since we already ship a `.ocamlinit` file, wouldn't it make sense to include these lines there?

**Review by Zimmi48 — APPROVED** [review]
Looks better to me after this refactoring.

File: dev/include_dune (hunk 2 above)
Comment (ejgallego) [reply dev/include_dune]: IMHO not as the included file may have other uses; for example utop will install printers differently [and automatically once we annotate them].

**Comment by ejgallego** [comment]
Failure is spurious.

**Comment by ejgallego** [comment]
Thanks @Zimmi48

**Pull request merged**